graviti/schema/pyarrow.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+"""The PyArrow extension type to represent PortexType."""
+
+import json
+from typing import Any, Optional
+
+import pyarrow as pa
+
+
+class PyArrowType(pa.ExtensionType):  # type: ignore[misc]  # pylint: disable=too-few-public-methods
+    """This class defines the PyArrow representation of PortexType type.
+
+    Arguments:
+        portex_type: The PortexType name.
+        repo: The repo of the PortexType, including the revision.
+        storage_type: The corresponding PyArrow type.
+        kwargs: Other keyword arguments to init the PortexType.
+
+    """
+
+    def __init__(
+        self, portex_type: str, storage_type: Any, repo: Optional[str] = None, **kwargs: Any
```

Comment:
1. `portex_type` is a `str`? Should it be `portex_type_name`?
2. Shall split `repo` into `url` and `revision`?

Reply:
1. Why `repo` is `Optional`?
2. shall we add `nullable` parameter?

Reply:
Why `storage_type` is `Any`?

Reply:
Numeric builtin types do not have repo.

Reply:
shall we another class for builtin type?